components/brave_wallet_ui/components/desktop/wallet-menus/asset-item-menu.tsx

```diff
@@ -0,0 +1,230 @@
+// Copyright (c) 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import * as React from 'react'
+import { useHistory } from 'react-router'
+import { skipToken } from '@reduxjs/toolkit/query/react'
+
+// Types
+import {
+  BraveWallet,
+  SendPageTabHashes,
+  WalletRoutes
+} from '../../../constants/types'
+
+// Queries
+import {
+  useGetOnRampAssetsQuery,
+  useGetVisibleNetworksQuery
+} from '../../../common/slices/api.slice'
+
+// Hooks
+import {
+  useMultiChainSellAssets
+} from '../../../common/hooks/use-multi-chain-sell-assets'
+
+// Utils
+import { getLocale } from '../../../../common/locale'
+import { getBalance } from '../../../utils/balance-utils'
+import {
+  useBalancesFetcher
+} from '../../../common/hooks/use-balances-fetcher'
+import Amount from '../../../utils/amount'
+
+// Components
+import {
+  SellAssetModal
+} from '../popup-modals/sell-asset-modal/sell-asset-modal'
+
+// Styled Components
+import {
+  StyledWrapper,
+  PopupButton,
+  PopupButtonText,
+  ButtonIcon
+} from './wellet-menus.style'
+
+interface Props {
+  asset: BraveWallet.BlockchainToken
+  assetBalance: string
+  account?: BraveWallet.AccountInfo
+}
+
+export const AssetItemMenu = (props: Props) => {
+  const {
+    asset,
+    assetBalance,
+    account
+  } = props
+
+  // routing
+  const history = useHistory()
+
+  // State
+  const [showSellModal, setShowSellModal] = React.useState<boolean>(false)
+
+  // Queries
+  const {
+    data: { allAssetOptions: allBuyAssetOptions } = {}
+  } = useGetOnRampAssetsQuery()
+  const { data: networkList = [] } = useGetVisibleNetworksQuery()
+  const {
+    data: tokenBalancesRegistry,
+  } = useBalancesFetcher(account && networkList
+    ? {
+      accounts: [account],
+      networks: networkList
+    }
+    : skipToken
+  )
+
+  // Hooks
+  const {
+    selectedSellAsset,
```

Comment:
Is `selectedSellAsset` ever different from the `asset` prop? If no, the `useBalancesFetcher` is not necessary.

Reply:
Good catch, removed and is now directly passing the `assetBalance` prop!